TankClient.java

```diff
@@ -91,442 +109,424 @@ public void framPaint(Graphics graphics) {
 				win=true;
 			}
 
-			if (homeTank.isLive() == false&&win==false) {
+			if (homeTank.isLive() == false&&homeTank2.isLive()==false&&win==false) {
 				Font f = graphics.getFont();
 				graphics.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 40));
 				tanks.clear();
 				bullets.clear();
 				graphics.drawString("Sorry. You lose!", 200, 300);
-				lose=true;
 				graphics.setFont(f);
-			}}else{
-				if (tanks.size() == 0 && home.isLive() && (homeTank.isLive()||homeTank2.isLive())&&lose==false) {
-					Font f = graphics.getFont();
-					graphics.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 60));
-					this.otherWall.clear();
-					graphics.drawString("Congratulations! ", 200, 300);
-					graphics.setFont(f);
-					win=true;
-				}
-
-				if (homeTank.isLive() == false&&homeTank2.isLive()==false&&win==false) {
-					Font f = graphics.getFont();
-					graphics.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 40));
-					tanks.clear();
-					bullets.clear();
-					graphics.drawString("Sorry. You lose!", 200, 300);
-					graphics.setFont(f);
-					lose=true;
-				}
-			}
-			graphics.setColor(c);
-
-			for (River r : theRiver) {
-				r.draw(graphics);
-			}
-
-			for (int i = 0; i < theRiver.size(); i++) {
-				River r = theRiver.get(i);
-				homeTank.collideRiver(r);
-				if(player2) homeTank2.collideRiver(r);
-				r.draw(graphics);
+				lose=true;
 			}
+		}
+		graphics.setColor(c);
 
-			home.draw(graphics);
-			homeTank.draw(graphics);
-			homeTank.eat(blood);
-			if (player2) {homeTank2.draw(graphics);homeTank2.eat(blood);}
-
-			for (int i = 0; i < bullets.size(); i++) {
-				Bullets m = bullets.get(i);
-				m.hitTanks(tanks);
-				m.hitTank(homeTank);
-				m.hitTank(homeTank2);
-				m.hitHome();
-				for(int j=0;j<bullets.size();j++){
-					if (i==j) continue;
-					Bullets bts=bullets.get(j);
-					m.hitBullet(bts);
-				}
-				for (MetalWall mw : metalWall) {
-					m.hitWall(mw);
-				}
+		for (River r : theRiver) {
+			r.draw(graphics);
+		}
 
-				for (CommonWall w : otherWall) {
-					m.hitWall(w);
-				}
+		for (int i = 0; i < theRiver.size(); i++) {

```

Comment:
For-loop šiuo atvėju galima patobulinti, parašant sąlygą taip pat, kaip ir #124 eilutėje. Tai pagerins kodo skaitomumą.